src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -314,6 +339,22 @@ public function loadMorphCount($relation, $relations)
         return $this;
     }
 
+    /**
+     * Enable relation autoload for the collection.
+     *
+     * @return $this
+     */
+    public function withRelationAutoload()
+    {
+        $callback = fn ($path) => $this->loadMissingRelationWithTypes($path);
+
+        $this
+            ->filter(fn ($model) => ! $model->hasRelationAutoloadCallback())
+            ->each(fn ($model) => $model->usingRelationAutoloadCallback($this, $callback));
```

Comment:
`filter+each` is an `O(2n)` operation; i'd suggest to do this in one each loop to keep things efficient, e.g.:

```php
->each(fn ($model) => ! $model->hasRelationAutoloadCallback() && $model->usingRelationAutoloadCallback($this, $callback));
```